tds/src/main/java/thredds/server/wfs/WFSController.java

```diff
@@ -96,27 +96,26 @@ private void describeFeatureType(PrintWriter out, HttpServletRequest hsreq) {
 	 * @param out
 	 * @return
 	 */
-	private void getFeature(PrintWriter out, HttpServletRequest hsreq, SimpleGeometryCSBuilder sgcs, String ftName) {
+	private WFSExceptionWriter getFeature(PrintWriter out, HttpServletRequest hsreq, SimpleGeometryCSBuilder sgcs, String ftName, String fullFtName) {
 		
 		List<SimpleGeometry> geometryList = new ArrayList<SimpleGeometry>();
 		
 		GeometryType geoT = sgcs.getGeometryType(ftName);
 		
-		switch(geoT) {
+		if(geoT == null){
+			return new WFSExceptionWriter("Feature Type of " + fullFtName + " not found.", "GetFeature" , "OperationProcessingFailed");

```

Comment:
Great error reporting! I like that the name of the missing feature type is included in the response.